src/main/kotlin/by/architecture_map/belarus/service/ConstructionImageService.kt

```diff
@@ -0,0 +1,14 @@
+package by.architecture_map.belarus.service
+
+import by.architecture_map.belarus.entity.ConstructionImage
+
+interface ConstructionImageService {
+
+    fun create(image: ConstructionImage): ConstructionImage
+    fun getRandomAndUniqueImages(gotImagesId: String?): MutableList<ConstructionImage>
+    fun getByConstructionArchitecturalStyleId(architecturalStyleId: Int): MutableList<ConstructionImage>
+
+    // If you can propose more descriptive name for this method please share <3
+    fun getImagesWithSameArchitecturalStyleByConstructionIdAcrossImagesOfCurrentConstruction(constructionId: Int): MutableList<ConstructionImage>
```

Comment:
`getImagesWithSameArchitecturalStyleByConstructionIdAcrossImagesOfCurrentConstruction`  - definitely this is not acceptable method name. do not try to describe what method does in the name. we have method comment for that JavaDoc in case of Java and KDoc in Kotlin

I can suggest `getSimilarImages` or something like that